src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -51,6 +52,18 @@ export const Configurator = ({
     return languages[0];
   });
 
+  const requiresTimeField = (ds: Dataset, lang: string, fields: DatasetField[]) => {
+    if (
+      languageService.getLanguage(lang)?.noTimeField ||
+      ds.type === DEFAULT_DATA.SET_TYPES.INDEX_PATTERN
+    ) {
+      return false;
+    }
+    if (fields && fields.length > 0) {
```

Comment:
what if they have no fields of type `date`? i think it's right though and matches the behavior othe index pattern management page that forces them still to click `i dont want to use a time filter`

Reply:
Should return false if no fields of type `date`, means there isn't any fields for the date picker to use.

Reply:
true but i think we might want to still get the user to select that they will have no fields to use.

i've seen OpenSearch automatically mapped a timestamp value to a string, so if we hide the timefield selector if we don't find any date fields I can see that being confusing if the user expects that they should have a timestamp field. but if they have a dropdown that provides no option besides that they dont want a time filter then it will provide a little bit more insight to the user that they are choosing to not aggregate their data on a timestamp